### PR TITLE
Edit modeling & asynchronous documentation to recommend Async.

### PR DESCRIPTION
**Pull request by sarahjohanknecht**
**Modeling docs:** 
 -Add information about async templates available for use.

**Async docs:**
 -Refactor documentation to recommend Task/Async templates, since Promise template is deprecated.

**Testing:**
 -jekyll serve to check formatting & links.

### CODE REVIEW

File: Asynchronous-Servers-and-Clients-in-Rest_li.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ There are two main options available to write async server implementations in Re
 
 ### Async Server Templates
 
-Rest.li also includes templates to make writing async resources slightly easier. There are templates that use `Callback`s, `Promise`s, and `Task`s for each type of Rest.li resource. For example, for collection resources with primitive keys we have `com.linkedin.restli.server.resources.CollectionResourceAsyncTemplate`, `com.linkedin.restli.server.resources.CollectionResourceTaskTemplate`, and `com.linkedin.restli.server.resources.CollectionResourcePromiseTemplate`. There are similar templates for complex key resources, association resources, and simple resources.
+Rest.li also includes templates to make writing async resources slightly easier. There are templates that use `Callback`s and `Task`s for each type of Rest.li resource. For example, for collection resources with primitive keys we have `com.linkedin.restli.server.resources.CollectionResourceAsyncTemplate` and `com.linkedin.restli.server.resources.CollectionResourceTaskTemplate`. There are similar templates for complex key resources, association resources, and simple resources.
```
Comment (maximelamure) [inline Asynchronous-Servers-and-Clients-in-Rest_li.md]: I might be wrong, but I think we recommended to use the `task` one.  Can you check? if yes, we can explicitly mention that.

Comment (sarahjohanknecht) [reply Asynchronous-Servers-and-Clients-in-Rest_li.md]: The Promise template is deprecated, so I am decreasing the mentions of that in this document. From the documentation on the templates, it doesn't seem that we specifically recommend task over async. The main difference is whether the user wants to use Parseq and Tasks or a callback. I can change the documentation to specifically recommend Task if thats what we would like to do going forward.

File: modeling.md [hunk 2]
```diff
@@ -171,18 +175,28 @@ keys are “foreign” keys from the referenced entities. Instead of using
...
+public class MyRelationResource extends AssociationResourceTaskTemplate<Relation> { … }  
 ```
 
 Defines a child resource with a URI of the form:
 
 ```  
 /myRelations/key1={longKey1}\&key2={longKey2}  
 ```
+#### Asynchronous resources
+Asynchronous Rest.li resources should be used when the downstream implementations are non-blocking. There are asynchronous resource templates for Collection, Simple, Association, and ComplexKey resources.
```
Comment (evanw555) [inline modeling.md]: What do you mean "when the downstream implementations are non-blocking"? Like if a service downstream from your resource uses an async template? May need a little more clarification

File: modeling.md [hunk 3]
```diff
@@ -171,18 +175,28 @@ keys are “foreign” keys from the referenced entities. Instead of using
 Create, new association relationships are made by using Update, which
 allows the client to provide the resource identifier.
 
+Asynchronous association resources are declared by creating a class that extends AssociationResourceAsyncTemplate or AssociationResourceTaskTemplate.
```
Comment (evanw555) [inline modeling.md]: Class names and other inline code snippets should be wrapped with graves (``) to make it look nice, (e.g. DataSchema -> `DataSchema`). GitHub's highlighting logic might be a little complicated, so idk if you've already done this. I don't see the graves but it's highlighted.

File: modeling.md (hunk 2 above)
Comment (sarahjohanknecht) [reply modeling.md]: I added a little more clarification. The page I linked at the bottom of these section goes more into detail about asynchronous, as well.

File: modeling.md (hunk 3 above)
Comment (sarahjohanknecht) [reply modeling.md]: Fixed! This file wasn't using that formatting for other class names of the templates, so I changed all of them to be formatted with graves.

**Comment by sarahjohanknecht** [comment]
I think changing ServiceErrors was outside of the scope.

**Pull request merged**